### PR TITLE
add code coverage

### PR DESCRIPTION
**Pull request by ShellyDCMS**
Add Cypress code coverage using Istanbul nyc
image

### CODE REVIEW

**Comment by HarelM** [comment]
Please add output file to fit ignore (out.json).

**Comment by HarelM** [comment]
Can you grant me permissions to your repo?
I'll fix the package lock and add the codecov CI step.

**Comment by HarelM** [comment]
I have fixed the package lock but it seems that the coverage is causing the test to fail for some reason.
I was able to reproduce this locally in codespaces when running the tests using command line (headless).
I also found this closed issue, which sounds similar, but it looks like it was fixed in latest version, so I don't know:
https://github.com/cypress-io/code-coverage/issues/216

**Comment by HarelM** [comment]
THANKS BTW!

**Comment by ShellyDCMS** [comment]
> I have fixed the package lock but it seems that the coverage is causing the test to fail for some reason. I was able to reproduce this locally in codespaces when running the tests using command line (headless). I also found this closed issue, which sounds similar, but it looks like it was fixed in latest version, so I don't know: [cypress-io/code-coverage#216](https://github.com/cypress-io/code-coverage/issues/216)
> 
> image

Is this still an issue? it seems we just need to add baseUrl

**Comment by HarelM** [comment]
Yes, this is still an issue, you can see it in the Firefox CI run.

**Comment by ShellyDCMS** [comment]
> Yes, this is still an issue, you can see it in the Firefox CI run.

Fixed

File: .nycrc.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "all": true,
+  "extends": "@istanbuljs/nyc-config-typescript",
+  "check-coverage": false,
+  "include": ["src/**/*.ts", "src/**/*.tsx"],
+  "exclude": [
+    "cypress/**/*.*",
```
Comment (HarelM) [inline .nycrc.json:7]: Can we remove some of the lines here? seems a bit too verbose...

**Review by HarelM — APPROVED** [review]
THANKS!

File: .nycrc.json (hunk 1 above)
Comment (HarelM) [reply .nycrc.json:7]: nit picking, this can be merged with and without this fix.

**Comment by HarelM** [comment]
Nice (besides the low number obviously)!
image

**Pull request merged**